db/migrate/20240124135559_categories.rb

```diff
@@ -0,0 +1,13 @@
+class Categories < ActiveRecord::Migration[7.0]
+  def change
+    create_table :categories do |t|
+      t.string :name, null:false, index: { unique:true }
+      t.timestamps
+    end
+
+    create_join_table :recipes, :categories do |t|
+      t.index :recipe_id
+      t.index :category_id
+    end
```

Comment:
One thing that I would add is a unique index on category_id + recipe_id.  This will prevent someone from adding the same category to a recipe more than once.

```suggestion
    end
    
    add_index :categories_recipes, [:category_id, :recipe_id], unique: true
```